firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -215,6 +239,23 @@ private void ensureClientConfigured() {
     }
   }
 
+  private FirebaseFirestoreSettings mergeEmulatorSettings(
+      @NonNull FirebaseFirestoreSettings settings,
+      @NonNull EmulatedServiceSettings emulatorSettings) {
+
+    if (!FirebaseFirestoreSettings.DEFAULT_HOST.equals(settings.getHost())) {
+      throw new IllegalStateException(
+          "Cannot change the Firestore host through FirebaseFirestoreSettings when "
+              + "EmulatedServiceSettings are also in effect. "
+              + "Make sure to only set the host in one location.");
+    }
```

Comment:
How about deprecating `FirebaseFirestoreSettings.setHost()` in favor of  `EmulatedServiceSettings`? Does it have any other use case other than pointing firestore to use the emulator as host?

Reply:
Yeah it's heavily used by the Firestore team themselves for testing against development/staging backends.

Reply:
@samtstern oh, ok. I understand.